diff --git a/index.d.ts b/index.d.ts
index 52d6365d5086b354c1cfa263047c55d8d07d08fd..0f090f957c7a047ecfbf6cb28628673ffbf05d89 100644
--- a/index.d.ts
+++ b/index.d.ts
@@ -14,7 +14,7 @@
 //                 Nandor Kraszlan <https://github.com/nandi95>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
-/// <reference types="node" />
+/// <reference types="node/buffer" />
 
 export class JsonWebTokenError extends Error {
     inner: Error;